src/main/java/net/minecraftforge/common/util/MutableHashedLinkedList.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+package net.minecraftforge.common.util;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.jetbrains.annotations.Nullable;
+
+import it.unimi.dsi.fastutil.Hash.Strategy;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenCustomHashMap;
+
+public class MutableHashedLinkedList<K, V> implements Iterable<Map.Entry<K, V>>
```

Comment:
Upon further reflection (in my IDE, while looking at the class more closely), this should be renamed to `MutableHashedLinkedMap` as this is not a list, but a map.

Here are also my suggested javadocs:
```suggestion
/**
 * A mutable linked map with a hashing strategy and a merge function.
 *
 * @param <K> the type of keys
 * @param <V> the type of mapped values
 */
public class MutableHashedLinkedMap<K, V> implements Iterable<Map.Entry<K, V>>
```